backend/coins/eth/etherscan/etherscan.go

```diff
@@ -457,6 +458,48 @@ func (etherScan *EtherScan) Balance(ctx context.Context, account common.Address)
 	return balance, nil
 }
 
+// Balances returns the balances for multiple addresses.
+func (etherScan *EtherScan) Balances(ctx context.Context, accounts []common.Address) (map[common.Address]*big.Int, error) {
+	if len(accounts) == 0 {
+		return nil, nil
+	}
+
+	params := url.Values{}
+	params.Set("module", "account")
+	params.Set("action", "balancemulti")
+
+	addresses := make([]string, len(accounts))
+	for i, account := range accounts {
+		addresses[i] = account.Hex()
+	}
+
+	params.Set("address", strings.Join(addresses, ","))
```

Comment:
https://docs.etherscan.io/api-endpoints/accounts

> up to 20 addresses per call

You should batch in chunks of 20 addresses then.